img_read_stream_survey: use separator

speleo3 opened the pull request:
Use `pimg->separator` with survey filtering in `img_read_stream_survey`.

Example:
```
*set separator /
*set names .
*begin foo
  *begin bar
    1 2 10 0 0
    2 3 5 90 0
  *end bar
*end foo
```
```sh
dump3d data.3d --survey=foo
```
Without this patch:
```
TITLE "foo"
DATE "@1728209233"
DATE_NUMERIC 1728209233
VERSION 8
SEPARATOR '/'
--
MOVE 0.00 0.00 0.00
STOP
```
With this patch:
```
TITLE "foo"
DATE "@1728209233"
DATE_NUMERIC 1728209233
VERSION 8
SEPARATOR '/'
--
MOVE 0.00 0.00 0.00
LINE 0.00 10.00 0.00 [bar] STYLE=NORMAL
LINE 5.00 10.00 0.00 [bar] STYLE=NORMAL
NODE 5.00 10.00 0.00 [bar/3] UNDERGROUND
NODE 0.00 10.00 0.00 [bar/2] UNDERGROUND
NODE 0.00 0.00 0.00 [bar/1] UNDERGROUND
STOP
```

speleo3 committed:
img_read_stream_survey: use separator

Use pimg->separator with survey filtering

ojwb commented:
Thanks.  The first part makes sense, but moving the second chunk of code breaks the title handling in other cases - e.g. before:

```
$ dump3d tests/multisurvey.plt --survey='Z+'|head -n1
TITLE "Fake Passage"
```

but with your change it uses the cave description instead of the description from the `Z+` survey:

```
$ src/dump3d tests/multisurvey.plt --survey='Z+'|head -n1
TITLE "FAKE CAVE"
```

I can see this change fixes a problem with the filtering, but can you explain exactly what the problem is that it's fixing?

speleo3 commented:
I moved the chunk because the separator needs to be assigned, which happens on line 1385
https://github.com/ojwb/survex/blob/44681da10b5ca675c2244dbe4e000108060cb2ae/src/img.c#L1385

Ah, I missed the function call to `compass_plt_open` and that it does its own survey filtering. So I suppose that `.plf` format should be excluded from the generic filtering?

speleo3 commented:
I reverted the chunk move and adjusted `stn_included` instead. Filtering will be wrong if filtering with trailing non-dot separator (`foo/`) and title will be wrong if filtering for `foo/bar`.

ojwb commented:
> I reverted the chunk move and adjusted `stn_included` instead. Filtering will be wrong if filtering with trailing non-dot separator (`foo/`) and title will be wrong if filtering for `foo/bar`.

Thanks.  I've merged and fixed those cases.

Also added test coverage for your case and a couple of others.

speleo3 commented:
Great, thank you!
